src/AppBundle/Service/SurveyNotifierManager.php

```diff
@@ -0,0 +1,223 @@
+            $notification->setTimeNotificationSent(new \DateTime());
+            $this->em->persist($notification);
+
+            $user = $notification->getUser();
+            $identifier = $notification->getUserIdentifier();
+            $phoneNumber = $user->getPhone();
+            $validNumber = $this->smsSender->validatePhoneNumber($phoneNumber);
+            if (!$validNumber) {
+                $this->logger->alert("Kunne ikke sende schedule sms til *$user*\n Tlf.nr.: *$phoneNumber*");
+                continue;
+            }
+
+            $message =
+                "Hei, ".$notification->getUser()->getFirstName()."\n".
+                $customMessage."\n".
+                $this->router->generate(
+                    'survey_show_user_id',
+                    ['id' => $surveyid, 'userid'=>$identifier],
+                    RouterInterface::ABSOLUTE_URL
+                ) .
+                "\n\n" .
+                "Med vennlig hilsen\n" .
+                "Vektorevaluering";
+
+            $sms = new Sms();
+            $sms->setMessage($message);
+            $sms->setSender("Vektor");
+            $sms->setRecipients([$phoneNumber]);
+            $this->smsSender->send($sms);
+        }
+
+        $this->isAllSent($surveyNotifier);
+        $this->em->flush();
+    }
+
+    private function sendEmail(SurveyNotifier $surveyNotifier)
+    {
+        $surveyId = $surveyNotifier->getSurvey()->getId();
+        $content = $surveyNotifier->getEmailMessage();
+        $subject = $surveyNotifier->getEmailSubject();
+        foreach ($surveyNotifier->getSurveyNotifications() as $notification) {
+            if ($notification->isSent()) {
+                return;
+            }
+            $notification->setSent(true);
+            $notification->setTimeNotificationSent(new \DateTime());
+            $this->em->persist($notification);
+
+
+            $user = $notification->getUser();
+            $identifier = $notification->getUserIdentifier();
+            $email = $user->getEmail();
+
+            $message = (new \Swift_Message())
+                ->setFrom(array($this->fromEmail => 'Vektorprogrammet'))
+                ->setSubject($subject)
+                ->setTo($email)
+                ->setReplyTo($this->fromEmail)
+                ->setBody(
+                    $this->twig->render(
+                        'survey/email_notification.html.twig',
+                        array(
+                            'firstname' => $notification->getUser()->getFirstName(),
+                            'route' => $this->router->generate('survey_show_user_id', ['id' => $surveyId, 'userid'=>$identifier], RouterInterface::ABSOLUTE_URL),
+                            'content' => $content,
+
+                        )
+                    ),
+                    'text/html'
+                );
+            $this->mailer->send($message);
+        }
+
+        $this->isAllSent($surveyNotifier);
+        $this->em->flush();
```

Comment:
Av erfaring fra `AdmissionNotification` burde vi flushe en gang for hver mail som sendes ut. Hvis vi får en error halvveis gjennom sendingen får vi nemlig ikke lagret noen notifications. Resultatet blir da at neste gang vi sender ut, vil alle som mottok i første omgang, motta en ekstra mail.